sw/host/opentitanlib/src/util/status.rs

```diff
@@ -0,0 +1,105 @@
+    filename: String,
+    // Line in the file where this status is created.
+    line: u16,
+}
+
+impl StatusCreateRecord {
+    fn new(record: &ot_status_create_record_t) -> Result<StatusCreateRecord> {
+        // Fields that are unknown are compile time use special value.
+        const UNKNOWN_MODULE_ID : u16 = 0xffffu16;
+        const UNKNOWN_ARG : u16 = 0xffffu16;
+        const UNKNOWN_CODE : u8 = 0xffu8;
+
+        // A C string is an array of char so bindgen creates an array of i8, so we need to
+        // convert it to an array of u8. Furthermore, since it is a fixed-size array, it contains
+        // a lot of zeroes at the end which CString does not like and that we need to remove.
+        let filename = record.filename
+            .map(|c| c as u8)
+            .into_iter()
+            .take_while(|c| *c != 0u8)
+            .collect::<Vec<_>>();
+        let filename = CString::new(filename)
+            .context(format!("invalid filename in status create record {:?}", record.filename))?
+            .to_string_lossy()
+            .to_string();
+
+        return Ok(StatusCreateRecord {
+            module_id: Some(record.module_id).filter(|mod_id| mod_id != &UNKNOWN_MODULE_ID),
+            arg: Some(record.arg).filter(|arg| arg != &UNKNOWN_ARG),
+            code: Some(record.code).filter(|code| code != &UNKNOWN_CODE),
+            filename: filename,
+            line: record.line,
+        })
+    }
+}
+
+pub fn load_elf(elf_file: &PathBuf) -> Result<Vec<StatusCreateRecord>> {
+    let file_data = std::fs::read(elf_file)
+        .context(format!("Could not read ELF file {}.", elf_file.display()))?;
+    let elf = ElfBytes::<AnyEndian>::minimal_parse(file_data.as_slice())
+        .context(format!("Could not interpret ELF file {}", elf_file.display()))?;
+    // Try to find the .ot.status_create_record section
+    let shdr: SectionHeader = elf
+        // returns a Result<Option<SectionHeader>, ParseError>
+        .section_header_by_name(".ot.status_create_record")
+        .context("cannot parse ELF file")?
+        .context("ELF file should have a .ot.status_create_record section")?;
+    let (status_create_records, compr) = elf
+        .section_data(&shdr)
+        .context("cannot read .ot.status_create_record section data")?;
+    // Make sure section data is not compressed (this is sanity check).
+    if compr.is_some() {
+        bail!(".ot.status_create_record section should not be compressed");
+    }
+    // Make sure that the section size is a multiple of the record size.
+    const RECORD_SIZE : usize = std::mem::size_of::<ot_status_create_record_t>();
+    if status_create_records.len() % RECORD_SIZE != 0 {
+        bail!(format!(".ot.status_create_record section size ({}) is not a multiple of the ot_status_create_record_t size ({})",
+                        status_create_records.len(), RECORD_SIZE));
+    }
+    // Convertion is unsafe but since the structure is packed and contains only POD,
```

Comment:
yes that's correct, the structure is packed and all bit patterns are valid. It could be some values make no sense (i.e. the code's value is not in the list of valid codes) but that will be handled by a higher abstraction yet to written.